### PR TITLE
feat: [PL-32566]: fix queries for users, accounts, userInvites collection

### PR DESCRIPTION
**Pull request by bhavya181**
## Describe your changes
fix queries for users, accounts, userInvites collection that may cause issues while loading data with limits on mongo queries.

changed some queries, applied NO_LIMIT with iterators in case we need to use all the data
New rest method has been added to list accountId through paginated api and used that instead to listing all accounts

Tested both of them locally added UTs as well

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger utAll`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/46389)

### CODE REVIEW

**Review by ankit-harness — CHANGES_REQUESTED** [review]
> Reviewed 16 of 17 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46389)* status: 17 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[`400-rest/src/main/java/software/wings/service/impl/AccountServiceImpl.java` line 250 at r1](https://reviewable.io/reviews/harness/harness-core/46389#-NSuXS05-rYf5X8TLL9k:-NSuXS05-rYf5X8TLL9l:b-sytkhf) ([raw file](https://github.com/harness/harness-core/blob/546d92a46db1e9bdeada84c794dbcb849dd0cb38/400-rest/src/main/java/software/wings/service/impl/AccountServiceImpl.java#L250)):*
> ```Java
> @TargetModule(_955_ACCOUNT_MGMT)
> @BreakDependencyOn("io.harness.delegate.beans.Delegate")
> public class AccountServiceImpl implements AccountService {
> ```

How can we avoid returning all accounts at once here? NO_LIMIT here can lead to issues if accounts increase in number

**Review by bhavya181 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46389)* status: Some of the checks are too old. 17 file(s) to review. 1 change requests. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`400-rest/src/main/java/software/wings/service/impl/AccountServiceImpl.java` line 250 at r1](https://reviewable.io/reviews/harness/harness-core/46389#-NSuXS05-rYf5X8TLL9k:-NTD2TAODY9G6iNf5B7u:b-m9ixyv) ([raw file](https://github.com/harness/harness-core/blob/546d92a46db1e9bdeada84c794dbcb849dd0cb38/400-rest/src/main/java/software/wings/service/impl/AccountServiceImpl.java#L250)):*
<details><summary><i>Previously, ankit-harness (Ankit Singhal) wrote…</i></summary><blockquote>

How can we avoid returning all accounts at once here? NO_LIMIT here can lead to issues if accounts increase in number
</blockquote></details>

NO_LIMIT has been applied to methods that uses iterators or for some NG APIs that calls CG to get list of all accounts, we can paginate those rest apis, but again then we have to make multiple calls from NG to CG and then accumulate, its the same thing

**Review by ankit-harness — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46389)* status: Some of the checks are too old. 17 file(s) to review. 1 change requests. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`400-rest/src/main/java/software/wings/service/impl/AccountServiceImpl.java` line 1089 at r1](https://reviewable.io/reviews/harness/harness-core/46389#-NTD9XKz4KuRDTk6n3XW:-NTD9XKz4KuRDTk6n3XX:be7bh45) ([raw file](https://github.com/harness/harness-core/blob/546d92a46db1e9bdeada84c794dbcb849dd0cb38/400-rest/src/main/java/software/wings/service/impl/AccountServiceImpl.java#L1089)):*
> ```Java
> 
>   @Override
>   public List<Account> listHarnessSupportAccounts(Set<String> excludedAccountIds, Set<String> fieldsToBeIncluded) {
> ```

Lets fix usages of listHarnessSupportAccounts
Also, lets create a tracking ticket with details for Gateway usage of this method
___
*[`400-rest/src/main/java/software/wings/service/impl/AccountServiceImpl.java` line 1235 at r1](https://reviewable.io/reviews/harness/harness-core/46389#-NTD9Red208An2slWX5a:-NTD9Red208An2slWX5b:b3cyqbe) ([raw file](https://github.com/harness/harness-core/blob/546d92a46db1e9bdeada84c794dbcb849dd0cb38/400-rest/src/main/java/software/wings/service/impl/AccountServiceImpl.java#L1235)):*
> ```Java
> 
>   @Override
>   public List<AccountDTO> getAllAccounts() {
> ```

Lets fix usages of getAllAccounts

**Review by bhavya181 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46389)* status: 28 file(s) to review. 1 change requests. 3. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`400-rest/src/main/java/software/wings/service/impl/AccountServiceImpl.java` line 1089 at r1](https://reviewable.io/reviews/harness/harness-core/46389#-NTD9XKz4KuRDTk6n3XW:-NTIUk8y12wEgyorXIPM:b-vilc8z) ([raw file](https://github.com/harness/harness-core/blob/546d92a46db1e9bdeada84c794dbcb849dd0cb38/400-rest/src/main/java/software/wings/service/impl/AccountServiceImpl.java#L1089)):*
<details><summary><i>Previously, ankit-harness (Ankit Singhal) wrote…</i></summary><blockquote>

Lets fix usages of listHarnessSupportAccounts
Also, lets create a tracking ticket with details for Gateway usage of this method
</blockquote></details>

https://harness.atlassian.net/browse/PL-32624

all the usages has been fixed
___
*[`400-rest/src/main/java/software/wings/service/impl/AccountServiceImpl.java` line 1235 at r1](https://reviewable.io/reviews/harness/harness-core/46389#-NTD9Red208An2slWX5a:-NTIUojr8PXSGaTgXPtt:b-3plm2e) ([raw file](https://github.com/harness/harness-core/blob/546d92a46db1e9bdeada84c794dbcb849dd0cb38/400-rest/src/main/java/software/wings/service/impl/AccountServiceImpl.java#L1235)):*
<details><summary><i>Previously, ankit-harness (Ankit Singhal) wrote…</i></summary><blockquote>

Lets fix usages of getAllAccounts
</blockquote></details>

removed all usages of this method except migration ones

File: 332-ci-manager/service/src/main/java/io/harness/ci/telemetry/CiTelemetryPublisher.java [hunk 1]
```diff
@@ -87,8 +91,22 @@ public void recordTelemetry() {
     }
   }
 
-  List<String> getAllAccounts() {
-    return CGRestUtils.getResponse(accountClient.getAllAccountIds(true));
+  List<String> getAllAccountIds() {
+    int pageIndex = 0, pageSize = 10000;
+    List<String> accountIds = new ArrayList<>();
+    while (true) {
```
Comment (harshjain12) [inline 332-ci-manager/service/src/main/java/io/harness/ci/telemetry/CiTelemetryPublisher.java:97]: there should be some hard limit.
We should not use while true

Comment (harshjain12) [reply 332-ci-manager/service/src/main/java/io/harness/ci/telemetry/CiTelemetryPublisher.java:97]: Ideally this should be part of client and logic should be defined in each method

**Comment by bhavya181** [comment]
trigger utAll

**Comment by bhavya181** [comment]
trigger utAll

**Comment by bhavya181** [comment]
trigger utAll

**Review by ankit-harness — CHANGES_REQUESTED** [review]
> Reviewed 1 of 13 files at r3, 1 of 21 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46389)* status: 45 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[`120-ng-manager/src/main/java/io/harness/telemetry/NGTelemetryPublisher.java` line 84 at r7](https://reviewable.io/reviews/harness/harness-core/46389#-NTsVV983ZT7ASDLfQBy:-NTsVV983ZT7ASDLfQBz:b-4m1664) ([raw file](https://github.com/harness/harness-core/blob/3a1a9498906f628e07a1744d272741335992aaa1/120-ng-manager/src/main/java/io/harness/telemetry/NGTelemetryPublisher.java#L84)):*
> ```Java
> 
>   private String getAccountId() {
>     List<AccountDTO> accountDTOList = CGRestUtils.getResponse(accountClient.listAccounts(0, 2)).getResponse();
> ```

Can't we just fetch account ids here?
___
*[`945-account-mgmt/src/main/java/io/harness/account/AccountClient.java` line 133 at r7](https://reviewable.io/reviews/harness/harness-core/46389#-NTsW3ax60mgdYD34tG9:-NTsW3ax60mgdYD34tGA:b6ljrbm) ([raw file](https://github.com/harness/harness-core/blob/3a1a9498906f628e07a1744d272741335992aaa1/945-account-mgmt/src/main/java/io/harness/account/AccountClient.java#L133)):*
> ```Java
>   Call<RestResponse<Boolean>> isImmutableDelegateEnabled(@Query("accountId") String accountId);
> 
>   @GET(ACCOUNT_API + "/list")
> ```

We are reusing the same endpoint. Dependent services may break?
___
*[`945-account-mgmt/src/main/java/io/harness/account/utils/AccountUtils.java` line 63 at r7](https://reviewable.io/reviews/harness/harness-core/46389#-NTsUr6C3HitN5YcrSpL:-NTsUr6C3HitN5YcrSpM:b-i85zo2) ([raw file](https://github.com/harness/harness-core/blob/3a1a9498906f628e07a1744d272741335992aaa1/945-account-mgmt/src/main/java/io/harness/account/utils/AccountUtils.java#L63)):*
> ```Java
>   }
> 
>   public List<String> getAccountIds(boolean includeOnlyNGEnabledAccounts) {
> ```

Lets split it into 2 methods - 

getAccountIds
getNGAccountIds
___
*[`960-persistence/src/main/java/io/harness/beans/PageRequest.java` line 59 at r7](https://reviewable.io/reviews/harness/harness-core/46389#-NTsWWeP8lhnzcq_6HOC:-NTsWWeP8lhnzcq_6HOD:b-a3had2) ([raw file](https://github.com/harness/harness-core/blob/3a1a9498906f628e07a1744d272741335992aaa1/960-persistence/src/main/java/io/harness/beans/PageRequest.java#L59)):*
> ```Java
>   @Inject static int maxDocumentsToBeFetchedByMongoQueries;
>   private static Pattern searchField = Pattern.compile("search\\[[0-9]+]\\[field]");
>   private static int maxLimit = maxDocumentsToBeFetchedByMongoQueries;
> ```

private static final int MAX_LIMIT?
___
*[`960-persistence/src/main/java/io/harness/beans/PageRequest.java` line 152 at r7](https://reviewable.io/reviews/harness/harness-core/46389#-NTsWeGB6pVzEGQ1vqIF:-NTsWeGB6pVzEGQ1vqIG:bfsnwu0) ([raw file](https://github.com/harness/harness-core/blob/3a1a9498906f628e07a1744d272741335992aaa1/960-persistence/src/main/java/io/harness/beans/PageRequest.java#L152)):*
> ```Java
>    */
>   public void setLimit(String limit) {
>     this.limit = UNLIMITED.equals(limit) ? limit : String.valueOf(Math.min(Integer.parseInt(limit), maxLimit));
> ```

Who passes UNLIMITED?

**Review by bhavya181 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46389)* status: 45 file(s) to review. 1 change requests. 6. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`120-ng-manager/src/main/java/io/harness/telemetry/NGTelemetryPublisher.java` line 84 at r7](https://reviewable.io/reviews/harness/harness-core/46389#-NTsVV983ZT7ASDLfQBy:-NTsajzwEXRftuPcSo20:b-fb5jes) ([raw file](https://github.com/harness/harness-core/blob/3a1a9498906f628e07a1744d272741335992aaa1/120-ng-manager/src/main/java/io/harness/telemetry/NGTelemetryPublisher.java#L84)):*
<details><summary><i>Previously, ankit-harness (Ankit Singhal) wrote…</i></summary><blockquote>

Can't we just fetch account ids here?
</blockquote></details>

there is not method which just fetches accountId
___
*[`332-ci-manager/service/src/main/java/io/harness/ci/telemetry/CiTelemetryPublisher.java` line 97 at r4](https://reviewable.io/reviews/harness/harness-core/46389#gh-1172406241:-NTsardE5QNwFvh5MYPx:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/5d50039ea4e0a28aab3a583082ab62c94bed9ae6/332-ci-manager/service/src/main/java/io/harness/ci/telemetry/CiTelemetryPublisher.java#L97)):*
<details><summary><i>Previously, harshjain12 (Harsh Jain) wrote…</i></summary><blockquote>

Ideally this should be part of client and logic should not be defined in each caller
</blockquote></details>

Done.
___
*[`945-account-mgmt/src/main/java/io/harness/account/AccountClient.java` line 133 at r7](https://reviewable.io/reviews/harness/harness-core/46389#-NTsW3ax60mgdYD34tG9:-NTsaskeAm9HEwOpbb_u:b-m13kor) ([raw file](https://github.com/harness/harness-core/blob/3a1a9498906f628e07a1744d272741335992aaa1/945-account-mgmt/src/main/java/io/harness/account/AccountClient.java#L133)):*
<details><summary><i>Previously, ankit-harness (Ankit Singhal) wrote…</i></summary><blockquote>

We are reusing the same endpoint. Dependent services may break?
</blockquote></details>

we have removed that method itself, shouldn’t the build will fail, if its not available?
___
*[`945-account-mgmt/src/main/java/io/harness/account/utils/AccountUtils.java` line 63 at r7](https://reviewable.io/reviews/harness/harness-core/46389#-NTsUr6C3HitN5YcrSpL:-NTsb4KM95aLD2HDtS--:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/3a1a9498906f628e07a1744d272741335992aaa1/945-account-mgmt/src/main/java/io/harness/account/utils/AccountUtils.java#L63)):*
<details><summary><i>Previously, ankit-harness (Ankit Singhal) wrote…</i></summary><blockquote>

Lets split it into 2 methods - 

getAccountIds
getNGAccountIds
</blockquote></details>

Done.
___
*[`960-persistence/src/main/java/io/harness/beans/PageRequest.java` line 59 at r7](https://reviewable.io/reviews/harness/harness-core/46389#-NTsWWeP8lhnzcq_6HOC:-NTsb5gN0aL-9Rxb5NMq:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/3a1a9498906f628e07a1744d272741335992aaa1/960-persistence/src/main/java/io/harness/beans/PageRequest.java#L59)):*
<details><summary><i>Previously, ankit-harness (Ankit Singhal) wrote…</i></summary><blockquote>

private static final int MAX_LIMIT?
</blockquote></details>

Done.
___
*[`960-persistence/src/main/java/io/harness/beans/PageRequest.java` line 152 at r7](https://reviewable.io/reviews/harness/harness-core/46389#-NTsWeGB6pVzEGQ1vqIF:-NTsb6KVA926D0Q9SlpE:b-1bxrz2) ([raw file](https://github.com/harness/harness-core/blob/3a1a9498906f628e07a1744d272741335992aaa1/960-persistence/src/main/java/io/harness/beans/PageRequest.java#L152)):*
<details><summary><i>Previously, ankit-harness (Ankit Singhal) wrote…</i></summary><blockquote>

Who passes UNLIMITED?
</blockquote></details>

here UNLIMITED means page size as 2000

**Review by bhavya181 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46389)* status: 45 file(s) to review. 1 change requests. 3. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`945-account-mgmt/src/main/java/io/harness/account/AccountClient.java` line 133 at r7](https://reviewable.io/reviews/harness/harness-core/46389#-NTsW3ax60mgdYD34tG9:-NTxiHj55SFm1f1zMLuJ:bmghi2a) ([raw file](https://github.com/harness/harness-core/blob/3a1a9498906f628e07a1744d272741335992aaa1/945-account-mgmt/src/main/java/io/harness/account/AccountClient.java#L133)):*
<details><summary><i>Previously, ankit-harness (Ankit Singhal) wrote…</i></summary><blockquote>

Services get deployed independently. So, your changes may not together
</blockquote></details>

added that api again on cg rest side 
___
*[`960-persistence/src/main/java/io/harness/beans/PageRequest.java` line 152 at r7](https://reviewable.io/reviews/harness/harness-core/46389#-NTsWeGB6pVzEGQ1vqIF:-NTxiO4u024ddYqtbVIh:b1ii6q2) ([raw file](https://github.com/harness/harness-core/blob/3a1a9498906f628e07a1744d272741335992aaa1/960-persistence/src/main/java/io/harness/beans/PageRequest.java#L152)):*
<details><summary><i>Previously, ankit-harness (Ankit Singhal) wrote…</i></summary><blockquote>

So MAX_LIMIT greater than 2000 as no effect?
</blockquote></details>

MAX_LIMIT will always be greater than 2000, and we are taking min of both, so, this should be equal to 2000 only.
___
*[`960-persistence/src/main/java/io/harness/beans/PageRequest.java` line 594 at r10](https://reviewable.io/reviews/harness/harness-core/46389#-NTxQVtn9_yvAUr-usKr:-NTxiYVq4-tRz9O0FT19:b-iyrnry) ([raw file](https://github.com/harness/harness-core/blob/e19f63f53087f6b87158694f94377bd216397f63/960-persistence/src/main/java/io/harness/beans/PageRequest.java#L594)):*
<details><summary><i>Previously, ankit-harness (Ankit Singhal) wrote…</i></summary><blockquote>

Can't we just call setLimit here?
</blockquote></details>

No, that sets limit in another object

**Review by ankit-harness — CHANGES_REQUESTED** [review]
> Reviewed 1 of 17 files at r8, 1 of 1 files at r10, 9 of 9 files at r11, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46389)* status: 45 file(s) to review. 1 change requests. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`400-rest/src/main/java/software/wings/resources/AccountResourceNG.java` line 87 at r11](https://reviewable.io/reviews/harness/harness-core/46389#-NTxlvz98WcfFj48pJVE:-NTxlvz98WcfFj48pJVF:b-bsf0g9) ([raw file](https://github.com/harness/harness-core/blob/b772de46f2a9d5c10f1d71fcaff6cd475e88375e/400-rest/src/main/java/software/wings/resources/AccountResourceNG.java#L87)):*
> ```Java
> 
>   @GET
>   @Path("/list")
> ```

Can we mark it deprecated and file a ticket to remove this API
___
*[`400-rest/src/main/java/software/wings/service/impl/UserGroupServiceImpl.java` line 1100 at r11](https://reviewable.io/reviews/harness/harness-core/46389#-NTxm5pB6XiJNm2Eee4T:-NTxm5pB6XiJNm2Eee4U:bcgjvre) ([raw file](https://github.com/harness/harness-core/blob/b772de46f2a9d5c10f1d71fcaff6cd475e88375e/400-rest/src/main/java/software/wings/service/impl/UserGroupServiceImpl.java#L1100)):*
> ```Java
>       for (UserInvite invite : userInvites) {
>         invite.getUserGroups().removeIf(x -> x.getUuid().equals(userGroupId));
>         log.info("Removing user group id {} from userInvites for unlink SSO group flow in account {}", userGroupId,
> ```

Why this change?

**Review by bhavya181 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46389)* status: 47 file(s) to review. 1 change requests. 3. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`400-rest/src/main/java/software/wings/resources/AccountResourceNG.java` line 87 at r11](https://reviewable.io/reviews/harness/harness-core/46389#-NTxlvz98WcfFj48pJVE:-NU0oj4r-OnvtoTeQ4DW:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/b772de46f2a9d5c10f1d71fcaff6cd475e88375e/400-rest/src/main/java/software/wings/resources/AccountResourceNG.java#L87)):*
<details><summary><i>Previously, ankit-harness (Ankit Singhal) wrote…</i></summary><blockquote>

Can we mark it deprecated and file a ticket to remove this API
</blockquote></details>

Done.
___
*[`400-rest/src/main/java/software/wings/service/impl/UserGroupServiceImpl.java` line 1100 at r11](https://reviewable.io/reviews/harness/harness-core/46389#-NTxm5pB6XiJNm2Eee4T:-NU0ojZZ3DrDe27_aQpa:b-7t5qkc) ([raw file](https://github.com/harness/harness-core/blob/b772de46f2a9d5c10f1d71fcaff6cd475e88375e/400-rest/src/main/java/software/wings/service/impl/UserGroupServiceImpl.java#L1100)):*
<details><summary><i>Previously, ankit-harness (Ankit Singhal) wrote…</i></summary><blockquote>

Why this change?
</blockquote></details>

this has been added in develop branch, just showing here as merge conflict was resolved on this.

**Comment by bhavya181** [comment]
trigger commonchecks

**Comment by bhavya181** [comment]
trigger feature-build

**Comment by bhavya181** [comment]
trigger codebasehash, utAll

**Comment by bhavya181** [comment]
trigger pmd

File: 332-ci-manager/service/src/main/java/io/harness/ci/telemetry/CiTelemetryPublisher.java (hunk 1 above)
Comment (bhavya181) [reply 332-ci-manager/service/src/main/java/io/harness/ci/telemetry/CiTelemetryPublisher.java]: done

**Review by archit-harness — COMMENTED** [review]
> Reviewed 2 of 21 files at r4, 1 of 9 files at r11.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46389)* status: 44 file(s) to review. 1 change requests. 3. unresolved discussions. 0 of 2 approvals obtained.

**Review by archit-harness — DISMISSED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46389)* status: 44 file(s) to review. 1 change requests. 3. unresolved discussions. 1 of 2 approvals obtained.

File: 945-account-mgmt/src/main/java/io/harness/account/utils/AccountUtils.java [hunk 2]
```diff
@@ -54,4 +59,37 @@ public String generateAccountName(String email) {
...
+          pageResponse.getResponse().stream().map(AccountDTO::getIdentifier).collect(Collectors.toList()));
+      pageIndex++;
+    }
+    return accountIds;
+  }
+
+  public List<String> getAllNGAccountIds() {
+    int pageIndex = 0, pageSize = 10000;
+    List<String> accountIds = new ArrayList<>();
+    for (int i = 0; i < 100; i++) {
```
Comment (harshjain12) [inline 945-account-mgmt/src/main/java/io/harness/account/utils/AccountUtils.java:81]: let us log warning if i touches 80.

Or find total iterations via another api

Comment (bhavya181) [reply 945-account-mgmt/src/main/java/io/harness/account/utils/AccountUtils.java]: done

**Review by harshjain12 — APPROVED** [review]

**Review by ankit-harness — APPROVED** [review]
:lgtm:

> Reviewed 2 of 2 files at r12, 1 of 1 files at r13, 1 of 1 files at r15, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46389#-:-NU75GEo1nLsv_eC1QAn:bnfp4nl)* status: 44 file(s) to review. 3. unresolved discussions. 1 of 2 approvals obtained.

**Review by puthrayaharness — APPROVED** [review]

**Review by harshjain12 — APPROVED** [review]

**Pull request merged**